Project Console: Cannot "View Companies" on prod

## Summary

The project manager is unable to "View Companies" in their CLA group, the UI shows a spinner and nothing happens.

## Background

Project manager logs into project.lfcla.com
Project manager selects their project. 
Project manager scrolls to a CLA group and selects "View Companies"
The UI loads forever.

## Expected behavior

The UI should show a list of companies that have signed a CCLA for the CLA group.

## Screenshots

If applicable, add screenshots to help explain your problem.

![image](https://user-images.githubusercontent.com/46732999/72453489-29d59100-378d-11ea-9fe5-6073e862a527.png)

![image](https://user-images.githubusercontent.com/46732999/72453400-06124b00-378d-11ea-972d-8a8bc111fcef.png)



## Environment

Please complete the following information:

* Environment: DEV and PROD 
* Browser chrome


## Additional Context

```
16:39:56
[ERROR] TypeError: ("unsupported operand type(s) for &: '{0}' and '{1}'", 'Comparison', 'bool') Traceback (most recent call last):   File "/var/task/wsgi_handler.py", line 108, in handler     return serverless_wsgi.handle_request(wsgi_app, event, context)   File "/var/task/serverless_wsgi.py", line 173, in handle_request     response = Response.from_app(app, environ)   File "/var/task/werkzeug/wra
[ERROR] TypeError: ("unsupported operand type(s) for &: '{0}' and '{1}'", 'Comparison', 'bool')
Traceback (most recent call last):
  File "/var/task/wsgi_handler.py", line 108, in handler
    return serverless_wsgi.handle_request(wsgi_app, event, context)
  File "/var/task/serverless_wsgi.py", line 173, in handle_request
    response = Response.from_app(app, environ)
  File "/var/task/werkzeug/wrappers/base_response.py", line 287, in from_app
    return cls(*_run_wsgi_app(app, environ, buffered))
  File "/var/task/werkzeug/wrappers/base_response.py", line 26, in _run_wsgi_app
    return _run_wsgi_app(*args)
  File "/var/task/werkzeug/test.py", line 1119, in run_wsgi_app
    app_rv = app(environ, start_response)
  File "/var/task/hug/api.py", line 502, in api_auto_instantiate
    return module.__hug_wsgi__(*args, **kwargs)
  File "falcon/api.py", line 274, in falcon.api.API.__call__
    raise
  File "falcon/api.py", line 269, in falcon.api.API.__call__
    responder(req, resp, **params)
  File "/var/task/hug/api.py", line 356, in version_router
    request, response, api_version=api_version, **kwargs
  File "/var/task/hug/interface.py", line 946, in __call__
    raise exception
  File "/var/task/hug/interface.py", line 917, in __call__
    self.call_function(input_parameters), context, request, response, **kwargs
  File "/var/task/hug/interface.py", line 839, in call_function
    return self.interface(**parameters)
  File "/var/task/hug/interface.py", line 129, in __call__
    return __hug_internal_self._function(*args, **kwargs)
  File "/var/task/cla/routes.py", line 1002, in get_project_companies
    return cla.controllers.project.get_project_companies(project_id)
  File "/var/task/cla/controllers/project.py", line 282, in get_project_companies
    signature_reference_type='company')
  File "/var/task/cla/models/dynamo_models.py", line 1948, in get_signatures_by_project
    (SignatureModel.signature_signed == signature_signed) &
  File "/var/task/pynamodb/expressions/condition.py", line 29, in __and__
    self.__class__.__name__, other.__class__.__name__)
```

--- a/cla-backend/cla/models/dynamo_models.py
+++ b/cla-backend/cla/models/dynamo_models.py
@@ -1812,6 +1812,24 @@ class Repository(model_interfaces.Repository):
             ret.append(repo)
         return ret
 
+def create_filter(attributes, model):
+    """
+    Helper function that creates filter condition based on available attributes
+
+    :param attributes: attributes consisting of model attributes and values
+    :rtype attributes: dict
+    :param model: Model instance that handles filtering
+    :rtype model: pynamodb.models.Model
+    """
+    filter_condition = None
+    for key, value in attributes.items():
+        if not value:
+            continue
+        condition = getattr(model, key) == value
+        filter_condition = (
+            condition if not isinstance(filter_condition, Condition) else filter_condition & condition
+        )
+    return filter_condition
 
 class SignatureModel(BaseModel):  # pylint: disable=too-many-instance-attributes
     """
@@ -2287,18 +2305,25 @@ class Signature(model_interfaces.Signature):  # pylint: disable=too-many-public-
             signature_reference_id=None,
             signature_user_ccla_company_id=None,
     ):
+
+        signature_attributes = {
+            "signature_signed": signature_signed,
+            "signature_approved": signature_approved,
+            "signature_type": signature_type,
+            "signature_reference_type": signature_reference_type,
+            "signature_reference_id": signature_reference_id,
+            "signature_user_ccla_company_id": signature_user_ccla_company_id
+        }
+        filter_condition = create_filter(signature_attributes, SignatureModel)
+
         cla.log.info("Loading signature by project for project_id: %s", project_id)
         signature_generator = self.model.signature_project_index.query(
-            project_id, filter_condition=(SignatureModel.signature_reference_type == signature_reference_type) &
-                                         (SignatureModel.signature_reference_id == signature_reference_id) &
-                                         (SignatureModel.signature_type == signature_type) &
-                                         (SignatureModel.signature_user_ccla_company_id == signature_user_ccla_company_id) &
-                                         (SignatureModel.signature_signed == signature_signed) &
-                                         (SignatureModel.signature_approved == signature_approved)
+            project_id, filter_condition=filter_condition
         )
 
         cla.log.info("Loaded signature by project for project_id: %s", project_id)
         signatures = []
+
         for signature_model in signature_generator:
             signature = Signature()
             signature.model = signature_model
@@ -2325,8 +2350,8 @@ class Signature(model_interfaces.Signature):  # pylint: disable=too-many-public-
                                          (SignatureModel.signature_project_id == project_id) &
                                          (SignatureModel.signature_type == 'ccla') &
                                          (SignatureModel.signature_user_ccla_company_id.does_not_exist()) &
-                                         (SignatureModel.signature_signed == True) &
-                                         (SignatureModel.signature_approved == True)
+                                         (SignatureModel.signature_signed.exists()) &
+                                         (SignatureModel.signature_approved.exists())
         )
         signatures = []
         for signature_model in signature_generator:
@@ -2359,8 +2384,8 @@ class Signature(model_interfaces.Signature):  # pylint: disable=too-many-public-
                                       (SignatureModel.signature_project_id == project_id) &
                                       (SignatureModel.signature_type == 'cla') &
                                       (SignatureModel.signature_user_ccla_company_id == company_id) &
-                                      (SignatureModel.signature_signed == True) &
-                                      (SignatureModel.signature_approved == True)
+                                      (SignatureModel.signature_signed.exists()) &
+                                      (SignatureModel.signature_approved.exists())
         )
 
         signatures = []
@@ -2395,7 +2420,7 @@ class Signature(model_interfaces.Signature):  # pylint: disable=too-many-public-
         # Query returns all the signatures that the company has signed a CCLA for.
         # Loop through the signatures and retrieve only the project IDs referenced by the signatures.
         signature_generator = self.model.signature_reference_index.query(
-            company_id, SignatureModel.signature_signed == True
+            company_id, SignatureModel.signature_signed.exists()
         )
         project_ids = []
         for signature in signature_generator:

